core/server/services/stats/lib/mrr-stats-service.js

```diff
@@ -0,0 +1,154 @@
+const {DateTime} = require('luxon');
+
+class MrrStatsService {
+    constructor({db}) {
+        this.db = db;
+    }
+
+    /**
+     * Get the current total MRR, grouped by currency (ascending order)
+     * @returns {Promise<MrrByCurrency[]>}
+     */
+    async getCurrentMrr() {
+        const knex = this.db.knex;
+        const rows = await knex('members_stripe_customers_subscriptions')
+            .select(knex.raw(`plan_currency as currency`))
+            .select(knex.raw(`SUM(
+                CASE WHEN plan_interval = 'year' THEN
+                    plan_amount / 12
+                ELSE 
+                    plan_amount
+                END
+            ) AS mrr`))
+            .whereIn('status', ['active', 'unpaid', 'past_due'])
+            .where('cancel_at_period_end', 0)
+            .groupBy('plan_currency')
+            .orderBy('currency');
+
+        if (rows.length === 0) {
+            // Add a USD placeholder to always have at least one currency
+            rows.push({
+                currency: 'usd',
+                mrr: 0
+            });
+        }
+
+        return rows;
+    }
+
+    /**
+     * Get the MRR deltas for all days (from old to new), grouped by currency (ascending alphabetically)
+     * @returns {Promise<MrrDelta[]>} The deltas sorted from new to old
+     */
+    async fetchAllDeltas() {
```

Comment:
I thought part of the reason for switching to reverse-ordering was so that we didn't have to query ALL of the events to figure out MRR for e.g. last 30 days. Is that not the case?

Reply:
Yes, that was one of the advantages. The main reason was more to have a higher precision. I wanted to add the `limit` option to it earlier, but that sparked some discussion (whether or not to use a `filter` or limit, refs https://github.com/TryGhost/Team/issues/1458, which seemed to settle on the idea that a `limit` option makes the most sense with the addition of a filter in the future), so I hold of making any decisions on that. Currently it is in line with the members stats endpoint, so maybe we should create a separate PR that adds the `limit` option to both?

Reply:
Thinking further on this, limit might not be a good idea if we have multiple currencies.

Reply:
We don't need it right now, so let's track it and leave it for later 👍🏼

Reply:
I added a new issue [here](https://github.com/TryGhost/Team/issues/1499)